### PR TITLE
testing++: formatting, remove Prove, add sections (random checking)

### PR DESCRIPTION
**Pull request by vindarel**

### CODE REVIEW

**Comment by vindarel** [comment]
@sheepduke All good for you? @AlbertoEAF @alx-a Any review welcomed! (typos, wording, structure, style…)

File: testing.md [hunk 1]
```diff
@@ -3,22 +3,28 @@ title: Testing
...
 
 ## Testing with FiveAM
 
 FiveAM has 3 levels of abstraction: check, test and suite. As you may have guessed:
-1. A check is a single assertion that asserts a boolean value. 
-2. A test is the smallest runnable unit. A test case that may contain multiple checks. Any check failure leads to failure of the whole test.
-3. A suite is a collection of tests. When a suite is run, all tests inside would be performed. Suite allows paternity, which means that running a suite will run all the tests defined in it and in its children suites.
 
-A simplest code sample containing 3 basic blocks mentioned above can shown as below:
+1. A **check** is a single assertion that asserts a boolean value. The most used check is `is`, followed by an assertion of yours. For example, `(is (= 2 (+ 1 1)))`.
```
Comment (AlbertoEAF) [inline testing.md:23]: Too many times "assertion". Is it really a **check** in FiveAM? Or can we call it simply an assertion?

```suggestion
1. A **check** or assertion. The most used check is `is`, which asserts that its single argument is truthy. For example, `(is (= 2 (+ 1 1)))`.
```

File: testing.md [hunk 2]
```diff
@@ -49,6 +58,12 @@ the function will return NIL in that case."
     (t nil)))
 ~~~
 
+We will test that code:
+
+- we must ensure that the content read in a file is the expected content,
+- we must test that the condition is signaled if the file doesn't exist.
```
Comment (AlbertoEAF) [inline testing.md:64]: Sorry I don't understand this paragraph. This is what we'll test below?

If so:
```suggestion
Next, our test code shall:
- ensure that the content read in a file is the expected content,
- test that the condition is signaled if the file doesn't exist.
```

still, I'm a bit lost

File: testing.md [hunk 3]
```diff
@@ -59,75 +74,97 @@ the function will return NIL in that case."
...
-Here a new suite named `read-file-as-string` has been defined. It is declared to be a child suite of `my-system` as specified by the `:in` keyword. The macro `in-suite` sets it as the default suite for test defined later.
+Here a new suite named `read-file-as-string` has been defined. It is declared to be a child suite of `my-system` as specified by the `:in` keyword. The macro `in-suite` sets it as the default suite for the tests defined later.
 
 ### Defining tests
 
-Before diving into tests, here is a brief introduction of checks you may use inside tests:
+Before diving into tests, here is a brief introduction of the available checks you may use inside tests:
 
-* The macro `is` might be the mostly used check. It simply checks if given expression returns a true value and generates a `test-passed` or `test-failure` result accordingly.
+* The macro `is` might be the mostly used check. It simply checks if the given expression returns a true value and generates a `test-passed` or `test-failure` result accordingly.
```
Comment (AlbertoEAF) [inline testing.md:120]: "The macro `is` might be the most used check"

File: testing.md (hunk 1 above)
Comment (vindarel) [reply testing.md:23]: yes it is called a check in fiveam.

Comment (vindarel) [reply testing.md:23]: "1. A **check** is a single assertion that checks that its argument is truthy. The most used check is `is`. For example, `(is (= 2 (+ 1 1)))`."

File: testing.md (hunk 2 above)
Comment (vindarel) [reply testing.md:64]: oh. mmh. "We will write tests for that code" (the code above). "In particular, we must ensure: that the content… - that a condition…".

File: testing.md (hunk 1 above)
Comment (AlbertoEAF) [reply testing.md:23]: Looks great ;)

File: testing.md (hunk 2 above)
Comment (AlbertoEAF) [reply testing.md:64]: Much clearer :+1:  :)

Comment (vindarel) [reply testing.md]: great, thanks :)

**Comment by sheepduke** [comment]
> @sheepduke All good for you? @AlbertoEAF @alx-a Any review welcomed! (typos, wording, structure, style…)
>
Although I am late for the party, but I would say I am more than satisfied with it.
I am really glad that you added more about the fixture thing. :-)

**Pull request merged**